src/customParser.ts

```diff
@@ -0,0 +1,330 @@
+    const geometryCount = fileReader.readUint32();
+    const attributeCount = fileReader.readUint32();
+    const byteCount = fileReader.readUint32();
+
+    const indexes = new FibonacciDecoder(asArrayBuffer, fileReader.location, byteCount);
+    fileReader.skip(byteCount);
+
+    geometryAttributes.push({'type': geometryType, 'indexes': indexes, 'count': geometryCount,
+    'attributeCount': attributeCount});
+
+    if (fileReader.location === asArrayBuffer.byteLength) {
+      break;
+    }
+  }
+
+  return {'magicBytes': magicBytes, 'formatVersion': formatVersion, 'optimizerVersion': optimizerVersion,
+  'arrayCount': arrayCount, 'arrays': arraysDictionary, 'geometries': geometryAttributes};
+}
+
+let nodeId: number;
+let treeIndex: number;
+let color: any;
+let x: number;
+let y: number;
+let z: number;
+let normal: THREE.Vector3;
+let deltaX: number;
+let deltaY: number;
+let deltaZ: number;
+let height: number;
+let radius: number;
+let radiusA: number;
+let radiusB: number;
+let angle: number;
+const center = new THREE.Vector3();
+const delta = new THREE.Vector3();
+
+const centerA = new THREE.Vector3();
+const centerB = new THREE.Vector3();
+
+function loadData(trueValuesArray: any, geometryInfo: any) {
+  nodeId =    geometryInfo.indexes.nextValue();
+  treeIndex = geometryInfo.indexes.nextValue();
+  color =     geometryInfo.indexes.nextValue();
+  const xIndex = geometryInfo.indexes.nextValue();
+  const yIndex = geometryInfo.indexes.nextValue();
+  const zIndex = geometryInfo.indexes.nextValue();
+
+  x =         trueValuesArray.x_values[xIndex];
+  y =         trueValuesArray.y_values[yIndex];
+  z =         trueValuesArray.z_values[zIndex];
+
+  center.set(x, y, z);
+
+  // Normal: All but spheres
+  if ([1, 3, 4].indexOf(geometryInfo.type) !== -1) {
+    normal = trueValuesArray.normals[geometryInfo.indexes.nextValue()];
+  }
+
+  // Delta: Box
+  if (geometryInfo.type === 1) {
+    deltaX = trueValuesArray.deltas[geometryInfo.indexes.nextValue()];
+    deltaY = trueValuesArray.deltas[geometryInfo.indexes.nextValue()];
+    deltaZ = trueValuesArray.deltas[geometryInfo.indexes.nextValue()];
+    delta.set(deltaX, deltaY, deltaZ);
+  }
+
+  // Height: Cylinders and sphere/ellipsoid segments
+  if ([3, 4].indexOf(geometryInfo.type) !== -1) {
+    height = trueValuesArray.heights[geometryInfo.indexes.nextValue()];
+  }
+
+  // Radius (one): Circle, sphere/ellipsoid segment, cylinder
+  if ([2, 4].indexOf(geometryInfo.type) !== -1) {
+    radius = trueValuesArray.radiuses[geometryInfo.indexes.nextValue()];
+  }
+
+  // Radiuses (two): Cones
+  if ([3].indexOf(geometryInfo.type) !== -1) {
+    radiusA = trueValuesArray.radiuses[geometryInfo.indexes.nextValue()];
+    radiusB = trueValuesArray.radiuses[geometryInfo.indexes.nextValue()];
+  }
+
+  // Angle: Box
+  if ([1].indexOf(geometryInfo.type) !== -1) {
+    angle = trueValuesArray.angles[geometryInfo.indexes.nextValue()];
+  }
+
+  // Matrix: ??
```

Comment:
What about these?

Reply:
I don't need them for box/sphere/circle, so they will be finished later